cmake/modules/ExtendPath.cmake

```diff
@@ -0,0 +1,13 @@
+function(extend_path joined_path base_path current_segment)
```

Comment:
This also (sorry comments out of order) seems to just be replicating existing llvm cmake files.